### PR TITLE
Update CHANGELOG for 5.8

### PR DESCRIPTION
**Pull request by yim-lee**

### CODE REVIEW

**Review by tomerd — APPROVED** [review]

**Review by neonichu — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -35,16 +35,32 @@ Swift 5.9
 Swift 5.8
 -----------
 
+* [SE-0378]
+
+  SwiftPM now supports token authentication when interacting with a package registry. The `swift package-registry` command has two new subcommands `login` and `logout` as defined in SE-0378 for adding/removing registry credentials.  
+
+* [#5810]
+
+// TODO
```
Comment (tomerd) [inline CHANGELOG.md:44]: SwiftPM now allows exposing an executable product that consists solely of a binary target that is backed by an artifact bundle. This allow vending binary executables as their own separate package, independently of the plugins that are using them.

Comment (tomerd) [inline CHANGELOG.md:44]:
```suggestion
SwiftPM now allows exposing an executable product that consists solely of a binary target that is backed by an artifact bundle. This allow vending binary executables as their own separate package, independently of the plugins that are using them.  
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -35,16 +35,32 @@ Swift 5.9
...
+
+* [#5819]
+
+// TODO
```
Comment (tomerd) [inline CHANGELOG.md:48]:
```suggestion

Improved handling of offline behavior when a cached version of a dependency exists on disk. SwiftPM will check for network availability status to determine if it should attempt to update a checked version of a dependency, and when offline will use the cached version without an update.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -35,16 +35,32 @@ Swift 5.9
...
+
+* [#5874]
+
+  In packages using tools version 5.8 or later, Foundation is no longer implicitly imported into package manifests. If Foundation APIs are used, the module needs to be imported explicitly.
+  
+* [#5892]
+
+// TODO
```
Comment (tomerd) [inline CHANGELOG.md:56]:
```suggestion
Expose the new `--emit-extension-block-symbols` and `--omit-extension-block-symbols` via  `swift package dump-symbol-graph`. `--emit-extension-block-symbols` dumps symbol graph files that extension block symbol format. The default behavior does not change. The `--omit-extension-block-symbols` flag will be used to explicitly disable the feature once the default behavior has been changed to `--emit-extension-block-symbols` in the future.
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -35,16 +35,32 @@ Swift 5.9
 Swift 5.8
 -----------
 
+* [SE-0378]
+
+  SwiftPM now supports token authentication when interacting with a package registry. The `swift package-registry` command has two new subcommands `login` and `logout` as defined in SE-0378 for adding/removing registry credentials.
```
Comment (tomerd) [inline CHANGELOG.md:40]:
```suggestion
* [SE-0362] 

  SwiftPM targets can now specify the upcoming language features they require. Package.swift manifest syntax has been expanded with an API to to include setting `enableUpcomingFeature` and `enableExperimentalFeature` flags at the target level, as specified by [SE-0362].
  
* [SE-0378]

  SwiftPM now supports token authentication when interacting with a package registry. The `swift package-registry` command has two new subcommands `login` and `logout` as defined in SE-0378 for adding/removing registry credentials.  
```

Comment (tomerd) [reply CHANGELOG.md:40]: cc @DougGregor

File: CHANGELOG.md [hunk 5]
```diff
@@ -35,16 +35,41 @@ Swift 5.9
...
+
+Improved handling of offline behavior when a cached version of a dependency exists on disk. SwiftPM will check for network availability status to determine if it should attempt to update a checked version of a dependency, and when offline will use the cached version without an update.
 
 * [#5874]
 
   In packages using tools version 5.8 or later, Foundation is no longer implicitly imported into package manifests. If Foundation APIs are used, the module needs to be imported explicitly.
+  
+* [#5892]
+
+Expose the new `--emit-extension-block-symbols` and `--omit-extension-block-symbols` via  `swift package dump-symbol-graph`. `--emit-extension-block-symbols` dumps symbol graph files that extension block symbol format. The default behavior does not change. The `--omit-extension-block-symbols` flag will be used to explicitly disable the feature once the default behavior has been changed to `--emit-extension-block-symbols` in the future.
```
Comment (MaxDesiatov) [inline CHANGELOG.md:65]: wording nit to make sure all items consistently use past tense

```suggestion
Added new `--emit-extension-block-symbols` and `--omit-extension-block-symbols` via  `swift package dump-symbol-graph`. `--emit-extension-block-symbols` dumps symbol graph files that extension block symbol format. The default behavior does not change. The `--omit-extension-block-symbols` flag will be used to explicitly disable the feature once the default behavior has been changed to `--emit-extension-block-symbols` in the future.
```

**Comment by tomerd** [comment]
@swift-ci smoke test

**Comment by tomerd** [comment]
thanks @yim-lee

**Pull request merged**